### PR TITLE
Add missing install deadcode

### PR DESCRIPTION
**Pull request by andriisoldatenko**
This PR fixes: deadcode command not found and as side effect fixing all linter issues.

### CODE REVIEW

File: pkg/util/conditions/service.go [hunk 1]
```diff
@@ -21,16 +21,6 @@ func SetServiceCreated(conditions *[]metav1.Condition, conditionType, name strin
 	_ = meta.SetStatusCondition(conditions, condition)
 }
 
-func SetServiceUpdated(conditions *[]metav1.Condition, conditionType, name string) {
```
Comment (gkrenn) [inline pkg/util/conditions/service.go:24]: `ServiceUpdatedReason` const is unused now

File: pkg/util/conditions/statefulset.go [hunk 2]
```diff
@@ -21,26 +21,6 @@ func SetStatefulSetCreated(conditions *[]metav1.Condition, conditionType, name s
 	_ = meta.SetStatusCondition(conditions, condition)
 }
 
-func SetStatefulSetUpdated(conditions *[]metav1.Condition, conditionType, name string) {
```
Comment (gkrenn) [inline pkg/util/conditions/statefulset.go:24]: Updated/Deleted consts are unused now

File: pkg/util/conditions/suffix.go [hunk 3]
```diff
@@ -17,10 +17,6 @@ func appendUpdatedSuffix(name string) string {
 	return fmt.Sprintf("%s %s", name, updatedSuffix)
 }
 
-func appendDeletedSuffix(name string) string {
```
Comment (gkrenn) [inline pkg/util/conditions/suffix.go:20]: delete suffix unused

Comment (andriisoldatenko) [reply pkg/util/conditions/suffix.go:20]: it's unclear what do you mean :)

**Review by aorcholski — DISMISSED** [review]

File: pkg/util/conditions/suffix.go (hunk 3 above)
Comment (andriisoldatenko) [reply pkg/util/conditions/suffix.go:20]: fixed in cd8132dc

File: pkg/util/conditions/statefulset.go (hunk 2 above)
Comment (andriisoldatenko) [reply pkg/util/conditions/statefulset.go:24]: fixed in cd8132dc

File: pkg/util/conditions/service.go (hunk 1 above)
Comment (andriisoldatenko) [reply pkg/util/conditions/service.go:24]: fixed in cd8132dc

File: hack/make/go.mk [hunk 4]
```diff
@@ -52,6 +52,6 @@ go/gen_mocks: prerequisites/mockery
 	mockery
 
 ## Runs deadcode https://go.dev/blog/deadcode
-go/deadcode:
```
Comment (gkrenn) [inline hack/make/go.mk:55]: Any reason why this is not called within `go/lint`?

Comment (andriisoldatenko) [reply hack/make/go.mk:55]: we are not using go/lint in CI, we use github action ... it was like this before I think.

Comment (gkrenn) [reply hack/make/go.mk:55]: That's ok, would still be nice if you call `go/deadcode` within the `go/lint` command so it automatically runs as soon as we lint

Comment (andriisoldatenko) [reply hack/make/go.mk:55]: do you mean this b0d4b8bf?

**Review by gkrenn — CHANGES_REQUESTED** [review]
Would still like to see the deadcode command to run with lint command, otherwise looks good to me. 

Maybe I wouldn't remove the deadcode from the utils, as this will be helpful in the future, but that's more personal opinion.

File: hack/make/go.mk (hunk 4 above)
Comment (gkrenn) [reply hack/make/go.mk:55]: Ups, didn't see that 👍

Comment (gkrenn) [reply hack/make/go.mk:55]: Well, that commit was on the wrong branch 😉

Comment (andriisoldatenko) [reply hack/make/go.mk:55]: fixed. :)

**Review by gkrenn — APPROVED** [review]

**Review by aorcholski — APPROVED** [review]

**Pull request merged**